### PR TITLE
Addition of optional language/model parameters to constraints (and download method adjustment)

### PR DESCRIPTION
**Pull request by alexander-zap**
# What does this PR do?

## Summary
*This PR adds optional language/model parameters to the `__init__` of multiple constraint. This enables usage of the constraints for other languages than English and even multilingual use cases.
Also it contains some changes of resource download logic.*

## Additions
- Optional language/model parameters for multiple constraints (previously hard-coded configurations are set as default values)

## Changes
- Download logic of resources has been changed (see `download_from_s3` method)

## Checklist
- [ X ] The title of your pull request should be a summary of its contribution.
- [ X ] Please write detailed description of what parts have been newly added and what parts have been modified. Please also explain why certain changes were made.
- [ X ] If your pull request addresses an issue, please mention the issue number in the pull request description to make sure they are linked (and people consulting the issue know you   are working on it)
- [ X ] To indicate a work in progress please mark it as a draft on Github.
- [ X ] Make sure existing tests pass.
- [ X ] Add relevant tests. No quality testing = no merge.
- [ X ] All public methods must have informative docstrings that work nicely with sphinx. For new modules/files, please add/modify the appropriate `.rst` file in `TextAttack/docs/apidoc`.'

### CODE REVIEW

**Comment by qiyanjun** [comment]
@alexander-zap we are having a bit delayed schedule here.. Will check and update with Comments.

**Comment by qiyanjun** [comment]
@alexander-zap please rebase

**Comment by qiyanjun** [comment]
@jinyongyoo @alexander-zap is this ready to merge?

**Comment by alexander-zap** [comment]
@jinyongyoo 
Yes, this is ready to merge 👍

**Comment by alexander-zap** [comment]
@qiyanjun Is anything else required for the merge?

**Comment by qiyanjun** [comment]
@alexander-zap sorry for slow response. Yoo, hanyu and I are all super busy recently. Need one review and merge

**Comment by alexander-zap** [comment]
@qiyanjun No worries, just wanted to ask. Thanks for the quick reply!

**Comment by alexander-zap** [comment]
@qiyanjun Any updates on the review?

File: textattack/shared/utils/install.py [hunk 1]
```diff
@@ -25,12 +25,13 @@ def s3_url(uri):
     return "https://textattack.s3.amazonaws.com/" + uri
 
 
-def download_if_needed(folder_name):
-    """Folder name will be saved as `.cache/textattack/[folder name]`. If it
+def download_from_s3(folder_name, skip_if_cached=True):
```
Comment (sanchit97) [inline textattack/shared/utils/install.py:28]: @alexander-zap @jinyongyoo  Is the intention of `skip_if_cached` flag to control whether to force re-download?

Comment (qiyanjun) [reply textattack/shared/utils/install.py:28]: @alexander-zap can you answer the question from @sanchit97?

**Comment by qiyanjun** [comment]
@cogeid can you run a thorough test of this PR .. then we merge.

File: textattack/shared/utils/install.py (hunk 1 above)
Comment (alexander-zap) [reply textattack/shared/utils/install.py:28]: This code was not written by me, so I sadly cannot answer this question. I think @jinyongyoo wrote this part of the code.

**Comment by qiyanjun** [comment]
@MintForever can you run and check this PR

**Comment by qiyanjun** [comment]
@sanchit97 can you fix the errors in this PR? and then a round of reviewing?

**Comment by cogeid** [comment]
@alexander-zap hey could you please rebase this, it would be really appreciated!

**Comment by sanchit97** [comment]
There has been a major update in #428, which might be reason for some conflicts.

**Comment by alexander-zap** [comment]
I pulled from origin/master and resolved all conflicts. Branch should now be up-to-date.

**Comment by qiyanjun** [comment]
@alexander-zap Thanks for the updates.. Sorry for the delay. @jinyongyoo recently did a large code-rework. Now our repo is stable! Please feel free to PR those augmentation changes you have added! We love to include them and share them with all textattack users!

**Comment by sanchit97** [comment]
This is ready to merge

**Review by qiyanjun — APPROVED** [review]
the only concern is the functional name change.. 
download_if_needed to download_from_s3 ... 

the change appears a minor name change. but it seems OK.

**Pull request merged**